kore/src/Kore/Internal/TermLike.hs

```diff
@@ -1435,6 +1440,68 @@ mkEvaluated
     -> TermLike variable
 mkEvaluated = updateCallStack . synthesize . EvaluatedF . Evaluated
 
+mkDefined
+    :: forall variable
+    .  HasCallStack
+    => InternalVariable variable
+    => TermLike variable
+    -> TermLike variable
+mkDefined = updateCallStack . Recursive.cata worker
```

Comment:
We should use `Recursive.unfold` here. When the input and output of an operation are both co-/recursive, we have the a choice to `fold` or `unfold`; the second one produces the result lazily (incrementally) so it composes better with other transformations.